# Project 1: Self-Tracking

Author: Shreya Parjan

Sep 30, 2019

This notebook 

# Table of contents


# Pre-Processing

In [1]:
import pandas as pd
import numpy as np

In [2]:
timeData = pd.read_csv("emailtimes.csv")
allData = pd.read_csv("allemails.csv")

In [7]:
timeData.head()

Day  Date  TotalTopics  TotalEmailInteractions  DigestEmailInteractions  \
0    1  9/13           50                      11                        1   
1    2  9/14           35                       4                        1   
2    3  9/15           37                       8                        0   
3    4  9/16           50                      13                        0   
4    5  9/17           47                      17                        1   

   0-3  3-6  6-9  9-12  12-15  15-18  18-21  21-24  
0    1    0    1    10     15      6      7      4  
1    0    0    2     7      5      9      4      2  
2    1    0    0    12      4     11      5      2  
3    3    0    0    10     10     10      5      9  
4    1    0    6     8     11      7      7      2

In [5]:
timeData.shape

(15, 13)

In [8]:
allData.head()

Subject Category
0  To the person who took my bike helmet... - 1 U...        t
1                   Check Out Our Website - 1 Update        e
2  FLOWER SUNDAY SALE, TOMORROW 12-4, CHAPEL STEP...        w
3  Auditions and Workshop for Absolutely Murder! ...        a
4  [The Wellesley Review] E-Board SOIs due TOMORR...        i

In [6]:
allData.shape

(491, 2)

In 2 weeks, I received 491 emails just to the Wellesley Class of 2021 Google group that I'm a part of!

# Individual Questions

## Exploration 1: How often do I check my email overall? Are there any day-to-day patterns for how often I check my inbox over 2 weeks?

In [31]:
import seaborn as sns
import plotly.express as px
import matplotlib.pyplot as plt

fig = px.bar(timeData, x='Date', y='TotalEmailInteractions',hover_data=['Date', 'TotalEmailInteractions','TotalTopics'], color='TotalTopics',
             labels={'Date':'Date','TotalEmailInteractions':'Total Inbox Interactions','TotalTopics':'Total Digest Emails'}, height=400)
fig.show()

## Exploration 2: How often do I interact with emails in digest?

In [103]:
import plotly.graph_objects as go
date=timeData['Date']
digest = timeData['DigestEmailInteractions']
total = timeData['TotalEmailInteractions']


fig = go.Figure(data=[
    go.Bar(name='Standard Email Visits', x=date, y=total),
    go.Bar(name='Digest Email Visits', x=date, y=digest),
])
# Change the bar mode
fig.update_layout(barmode='stack')
fig.show()

# Overarching Questions

## Exploration 3: What kind of emails are most often sent to all classes?

In [94]:
import plotly.graph_objects as go

letterCodes = ['g','a','c','m','p','s','i','o','e','r','w','t','u','x','j','v','y']
letterCounts = np.zeros(len(letterCodes))


for i in range(len(letterCodes)):
    letterCounts[i] = len(allData.loc[allData['Category'] == letterCodes[i]])

dictionary = dict(zip(letterCodes, letterCounts))

newKeys = ['College Gov.','Performing Arts','CoOp','Music','Politics','Society','Special Interest','Sports','Service','Resources','Campus-Wide','Missing Item','Cultural','Unknown','Job','Off-Campus Event','Survey']
final_dict = dict(zip(newKeys, list(dictionary.values()))) 
final_dict


from operator import itemgetter  

labels = []
values = []
for key, value in sorted(final_dict.items(), key = itemgetter(1), reverse = True):
    labels.append(key)
    values.append(value)    

import plotly.graph_objects as go


fig = go.Figure(data=[go.Pie(labels=labels[:10], values=values[:10], hole=.3)])
fig.show()

## Exploration 4: When is the "prime time" for spamming all students?

In [110]:
timeList = ['12-3 AM', '3-6 AM','6-9 AM','9-12 PM','12-3 PM','3-6 PM','6-9 PM','9-12 AM']
countList = [timeData['0-3'].sum(),timeData['3-6'].sum(),timeData['6-9'].sum(),timeData['9-12'].sum(),timeData['12-15'].sum(),timeData['15-18'].sum(),timeData['18-21'].sum(),timeData['21-24'].sum()]

fig = go.Figure([go.Bar(x=timeList, y=countList,marker_color='crimson')])
fig.show()
